sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -93,6 +93,23 @@ object IntervalUtils {
 
   private val yearMonthPattern = "^([+|-])?(\\d+)-(\\d+)$".r
 
+  private val yearMonthFuzzyPattern = "[^-|+]*?([+|-]?[\\d]+-[\\d]+).*".r
```

Comment:
why do we need a new regex?

Reply:
> why do we need a new regex?

Since  yearMonthPattern only support `[+|-]yy-mmmm`

Reply:
what do we want to support here?

Reply:
I am discussing  this  here https://github.com/apache/spark/pull/32266#discussion_r619231878 with  @MaxGekk , 
Do you have any other supplementary suggestions?